download_py_package.py

```diff
@@ -55,20 +55,36 @@ def download_py_package():
             version_exists = True
     else:
         if version_exists:
-            message_contents = provides_source.MessageContents(
-                package_name=Configuration.PACKAGE_NAME,
-                package_version=Configuration.PACKAGE_VERSION,
-                index_url=Configuration.PACKAGE_INDEX,
-                value=False,
-            )
-            provides_source.sync_publish_to_topic(message_contents)
+            message_contents = [
+                {
+                    "topic_name": UpdateProvidesSourceDistroMessage.topic_name,
+                    "message_contents": {
+                        "package_name": Configuration.PACKAGE_NAME,
+                        "package_version": Configuration.PACKAGE_VERSION,
+                        "index_url": Configuration.PACKAGE_INDEX,
+                        "value": False,
+                    },
+                }
+            ]
+            with open(MESSAGE_LOCATION, "w") as f:
+                content = json.dumps(message_contents, indent=4)
+                f.write(content)
+            raise Exception("Missing source distro (message being sent)")
```

Comment:
is this workflow task going to fail? I would not expect task to fail for a known reason, rather continue if the error is known. WDYT? So that if we see too many SI workflow failures is because something new is hapenning.

Reply:
I was planning on having success vs failure be the conditional but we could also write to another file which indicates success or failure of the step and use that for the conditional logic since it is expected behaviour

Reply:
Right but from a monitoring and operations point of view would be better if the task and therefore the workflow does not fail for known issues because we might trigger alarms that are not required. Thanks!